src/nvim/api/extmark.c

```diff
@@ -1291,3 +1281,47 @@ Boolean nvim__win_del_ns(Window window, Integer ns_id, Error *err)
 
   return true;
 }
+
+/// EXPERIMENTAL: this API will change in the future.
+///
+/// Set some properties for namespace
+///
+/// @param ns_id Namespace
+/// @param opts Optional parameters to set:
+///           - scoped: (boolean) Make the namespace a scoped namespace
+///
+void nvim__ns_set(Integer ns_id, Dict(ns_opts) *opts, Error *err)
```

Comment:
this should accept `opts.win` (or maybe `opts.wins: number[]` if we really care about avoiding multiple calls) to set the scope.

That allows us to remove the `nvim__win_xx_ns` functions, right?